charts/sn-platform/templates/proxy/proxy-cluster.yaml

```
@@ -57,8 +57,8 @@ spec:
 {{ toYaml . | indent 6 }}
 {{- end }}
 {{- end }}
+{{- if or .Values.proxy.readPublicKeyFromFile .Values.proxy.extraSecretRefs }}
     secretRefs:
-    {{- if .Values.proxy.readPublicKeyFromFile }}
     {{- if .Values.proxy.publicKeyPath }}
```

I've experienced bugs like this before. The root cause of not noticing them is usually that the code isn't clear.

For example, here are five lines (if, template, else, template, end) that could be this one line:

```suggestion
    - mountPath: {{ default "/pulsar/vault/v1/identity/oidc/.well-known/keys" .Values.proxy.publicKeyPath }}
```

The lines below this and the broker-cluster.yaml (in two spots) could similarly be cleaned up.

Again, my motivation being that simpler templating leads to cleaner templating. Cleaner templating makes writing bug-free templates easier.